xcdat/averager.py

```diff
@@ -353,11 +377,13 @@ def _averager(xvariable, weights, axis):
     return wavg
 
 
-def spatial_average(xvariable,
-                    lat_bounds=None,
-                    lon_bounds=None,
-                    axis=['lat', 'lon'],
-                    weight='weighted'):
+def spatial_average(
+    xvariable: xr.DataArray,
+    axis: Union[List[SupportedAxes], SupportedAxes],
```

Comment:
`axis` must be supplied

Reply:
> Is this in order to validate that the axes are in the list of SupportedAxes?

I think it is better to be explicit about which `axis` to average over. There is also defensive code to handle if the `axis` argument is incorrect.

> I noticed if I pass in a non-supported axis (e.g., `axis=['x', 'y']), I don't get an error. What is this doing if not checking the axis values passed in?

Not sure why you don't receive the `ValueError` because I do:
```
ValueError: Incorrect `axis` argument. Supported axis include: lat, lon.
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/Documents/Repositories/tomvothecoder/xcdat/qa/qa.py in 
      58 from xcdat.averager import spatial_average
      59 
----> 60 spatial_average(
      61     ts,
      62     axis=["x", "y"],

~/Documents/Repositories/tomvothecoder/xcdat/xcdat/averager.py in spatial_average(xvariable, axis, lat_bounds, lon_bounds, is_weighted)
    473     for axes in axis:
    474         if axes not in SUPPORTED_AXES:
--> 475             raise ValueError(
    476                 "Incorrect `axis` argument. Supported axis include: "
    477                 f"{', '.join(SUPPORTED_AXES)}."

ValueError: Incorrect `axis` argument. Supported axis include: lat, lon.
```

> We should update this to allow the user to specify a weighting array.

Got it, I will add this as a function param and update the algorithm accordingly.